marconi-core/test/Marconi/CoreSpec.hs

```diff
@@ -1633,3 +1635,106 @@ withRollbackFailureTest =
   Tasty.testProperty "Rollback failure in a worker exit nicely" $
     Test.withMaxSuccess 10_000 $
       checkOutOfRollback (Lens.view defaultChain <$> Test.arbitrary)
+
+-- * Query modification - Stability
+withStabilityTestGroup :: Tasty.TestTree
+withStabilityTestGroup =
+  Tasty.testGroup
+    "StabilityTests"
+    [ Tasty.testGroup
+        "Stability"
+        [ Tasty.testGroup
+            "stability tests"
+            [ Tasty.testProperty "all results are stable" allStable
+            , Tasty.testProperty "not all results are stable" notAllStable
+            , Tasty.testProperty "withStability will not alter the events" withStabilityUnchanged
+            , Tasty.testProperty "no results are stable at a specific point" notStableAt
+            , Tasty.testProperty "all results are stable at a specific point" stableAt
+            , Tasty.testProperty "withStabilityAt will not alter the events" withStabilityAtUnchanged
+            ]
+        ]
+    ]
+
+-- | Helper for common chain generation logic
+genCommonChain :: (Int -> TestPoint) -> Gen ([Item TestEvent], Int, Word, Int)
+genCommonChain testPoint' = Test.sized $ \n -> do
+  k <- Test.chooseInt (1, 10)
+  (chain :: [Item TestEvent]) <-
+    genChain $
+      GenChainConfig
+        (pure (if n < 0 then n else 1))
```

Comment:
you don't need to care about the negative number: `sized` can issue a negative number, despite the `Int` type.